src/collections/maple_tree/mod.rs

```diff
@@ -370,159 +505,185 @@ where
-            return ops::Range{start: (rangeB-2) as isize, end: R_CAPACITY as isize}; 
+        if rangeB == 1 {
+            return ops::Range {
+                start: -1,
+                end: R_CAPACITY as isize,
+            };
+        } else {
+            return ops::Range {
+                start: (rangeB - 2) as isize,
+                end: R_CAPACITY as isize,
+            };
         }
-
     }
 
-    pub fn move_pivots_left(&mut self, pivotIndex: usize, places: usize) -> bool{
-        
-        if pivotIndex >= R_CAPACITY || pivotIndex < 0 || places > R_CAPACITY{
+    pub fn move_pivots_left(&mut self, pivotIndex: usize, distance: usize) -> bool {
+        if pivotIndex >= R_CAPACITY || pivotIndex < 0 || distance > R_CAPACITY {
             return false;
         }
-        
-        // the new index for the pivot at pivotIndex is pivotIndex-places
+
+        // the new index for the pivot at pivotIndex is pivotIndex-distance
         // make sure this is not negative
-        if (pivotIndex as isize) - (places as isize) < 0{
+        if (pivotIndex as isize) - (distance as isize) < 0 {
             return false;
         }
-           
+
         let mut pivotHolder: M<K> = M::None;
         let mut valueHolder: M<V> = M::None;
-               
-        for i in pivotIndex..R_CAPACITY{
 
+        for i in pivotIndex..R_CAPACITY {
             mem::swap(&mut self.pivot[i], &mut pivotHolder);
             mem::swap(&mut self.value[i], &mut valueHolder);
 
-            mem::swap(&mut self.pivot[i-places], &mut pivotHolder);
-            mem::swap(&mut self.value[i-places], &mut valueHolder); 
-            
+            mem::swap(&mut self.pivot[i - distance], &mut pivotHolder);
+            mem::swap(&mut self.value[i - distance], &mut valueHolder);
+
             pivotHolder = M::None;
             valueHolder = M::None;
         }
 
+        //TODO: replace these magic numbers with variables
+        if R_CAPACITY - pivotIndex < distance {
+            for i in (pivotIndex - distance + 1) + (R_CAPACITY - pivotIndex) - 1..R_CAPACITY {
+                self.pivot[i] = M::None;
+                self.value[i] = M::None;
+                self.value[i + 1] = M::None;
+            }
+        }
+
         return true;
     }
-     
-    // moves the pivot/s at pivotIndex and above to the right x places where x = places variable
-    // if the move would push values off the end then false is returned 
-    pub fn move_pivots_right(&mut self, pivotIndex: usize, places: usize) -> bool{
-       
-        if places == 0{
+
+    // moves the pivot/s at pivotIndex and above to the right x distance where x = distance variable
+    // if the move would push values off the end then false is returned
+    pub fn move_pivots_right(&mut self, pivotIndex: usize, distance: usize) -> bool {
+        if distance == 0 {
             return true;
         }
-        let freePivots = R_CAPACITY-self.len();
+        let freePivots = R_CAPACITY - self.len();
```

Comment:
More a stylistic thing, but I always prefer free_pivots rather than freePivots? It's a style thing but I'm not gonna stress too much because it's an easy change for later.